src/Apps/Order/OrderApp.tsx

```diff
@@ -18,26 +18,55 @@ export interface OrderAppProps {
   location: any
 }
 
-export const OrderApp: SFC<OrderAppProps> = ({
-  me,
-  children,
-  location,
-  ...props
-}) => {
-  const stepIndex = getStepIndex(location.pathname)
-  return (
-    <>
-      <Row>
-        <Col>
-          <Stepper initialTabIndex={stepIndex} currentStepIndex={stepIndex}>
-            {steps.map(step => <Step name={step} key={step} />)}
-          </Stepper>
-        </Col>
-      </Row>
-
-      <Spacer mb={3} />
-
-      {children}
-    </>
-  )
+export class OrderApp extends React.Component<OrderAppProps> {
+  removeTransitionHook: () => void
+
+  constructor(props) {
+    super(props)
+
+    this.removeTransitionHook = props.router.addTransitionHook(
+      this.onTransition
+    )
+  }
+
+  componentWillUnmount() {
+    this.removeTransitionHook()
+  }
+
+  onTransition = location => {
+    // FIXME: This logic shouldn't be invoked on a non-form based route
+    // Refresh
+    if (window && location.pathname === window.location.pathname) {
+      // Most browsers will ignore this and supply their own messaging for refresh
+      return "Are you sure you want to refresh? Your changes will not be saved."
+    }
+
+    // https://regexper.com/#%5C%2Forder%5Cd%3F%5C%2F.%2B%5C%2F%28%5Cw%2B%29
+    // Attempting to navigate to another route
+    if (location.pathname.match(/\/order\d?\/.+\/(\w+)/)) {
```

Comment:
is it possible to scope this directly from the `router` object rather than regex?

Reply:
That is an _excellent_ suggestion. Thanks Chris! It felt ~a little~ dirty, but I've been staring at this forever, ha. I'll update it!